charts/delphai-deployment/templates/deployment.yaml

```diff
@@ -13,7 +13,9 @@ metadata:
     helm.sh/chart: "{{ $.Chart.Name }}-{{ $.Chart.Version | replace "+" "_" }}"
 
 spec:
+  {{ if not $.Values.rabbitmq.enabled }}
   replicas: {{ $.Values.replicas }}
+  {{ end }} {{/* if not $.Values.rabbitmq.enabled */}}
```

Comment:
Isn't that a required field?
And even if not — I wouldn't have this condition here to not add unnecessary conditions, to keep it simple

Reply:
When there is automatic horizontal scaling (through KEDA), we must not try to manage this scaling manually. KEDA will manage this for us, fighting KEDA and trying to overrule it will only bring us trouble.

Reply:
We won't manage it manually.
Not defining it here means default value of 1 is used. But we also have 1 as the default.

So defining or not defining it yields the same result in ETCD database.

Why to add 2 extra lines with no effect?

Reply:
I wrote this condition based on the following k8s documentation: https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#replicas

> If a [HorizontalPodAutoscaler](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/) (or any similar API for horizontal scaling) is managing scaling for a Deployment, don't set .spec.replicas.

> Instead, allow the Kubernetes [control plane](https://kubernetes.io/docs/reference/glossary/?all=true#term-control-plane) to manage the .spec.replicas field automatically.

Reply:
For the example `translation` service that has the manually-created KEDA ScaledObject, the original Helm values say to do `spec.replicas: 1`, but the actual Deployment as it exists in the review cluster has `spec.replicas: 2`. KEDA takes ownership of the `spec.replicas` field.

Reply:
Yeah, makes sense. Thanks for the explanation.
Lets keep that then